static/app/components/replays/videoReplayer.tsx

```diff
@@ -0,0 +1,398 @@
+    if (root) {
+      root.appendChild(this.wrapper);
+    }
+
+    this._videos = this._attachments.map((attachment, index) =>
+      this.createVideo(attachment, index)
+    );
+    this._trackList = this._attachments.map(({timestamp}, i) => [timestamp, i]);
+    this.loadSegment(0);
+  }
+
+  private createVideo(segmentData: VideoEvent, index: number) {
+    const el = document.createElement('video');
+    el.src = `${this._videoApiPrefix}${segmentData.id}/`;
+    el.style.display = 'none';
+
+    // TODO: only attach these when needed
+    el.addEventListener('ended', () => this.handleSegmentEnd(index));
+    el.addEventListener('play', event => {
+      if (index === this._currentIndex) {
+        this._callbacks.onLoaded(event);
+      }
+    });
+    el.addEventListener('loadedmetadata', event => {
+      // Only call this for current segment?
+      if (index === this._currentIndex) {
+        this._callbacks.onLoaded(event);
+      }
+    });
+    // TODO: Only preload when necessary
+    el.preload = 'auto';
+    // TODO: Timer needs to also account for playback speed
+    el.playbackRate = this.config.speed;
+
+    // Append the video element to the mobile player wrapper element
+    this.wrapper.appendChild(el);
+
+    return el;
+  }
+
+  private handleSegmentEnd(index: number) {
+    const nextIndex = index + 1;
+
+    // No more segments
+    if (nextIndex >= this._attachments.length) {
+      this._timer.stop();
+      this._callbacks.onFinished();
+      return;
+    }
+
+    this.playSegmentAtIndex(nextIndex);
+  }
+
+  /**
+   * Given a relative time offset, get the segment number where the time offset would be contained in
+   */
+  protected getSegmentIndexForTime(relativeOffsetMs: number): {
+    previousSegment: number | undefined;
+    segment: number | undefined;
+  } {
+    const timestamp = this._startTimestamp + relativeOffsetMs;
+
+    // This function will return the prior segment index if no valid segments
+    // were found, so we will need to double check if the result was an exact
+    // match or not
+    const result = findVideoSegmentIndex(this._trackList, this._attachments, timestamp);
+    const resultSegment = this.getSegment(result)!;
+    const isExactSegment =
+      resultSegment &&
+      timestamp >= resultSegment.timestamp &&
+      timestamp <= resultSegment.timestamp + resultSegment.duration;
+
+    // TODO: Handle the case where relativeOffsetMs > length of the replay/seekbar (shouldn't happen)
+    return {
+      segment: isExactSegment ? result : undefined,
+      previousSegment: !isExactSegment ? result : undefined,
+    };
+  }
+
+  protected getSegment(index?: number | undefined): VideoEvent | null {
+    if (typeof index === 'undefined') {
+      return null;
+    }
+
+    return this._attachments[index];
+  }
+
+  protected getVideo(index: number | undefined): HTMLVideoElement | null {
+    if (typeof index === 'undefined') {
+      return null;
+    }
+
+    return this._videos[index];
```

Comment:
This could be `.at(index)`, and/or suffixed with `?? null` explicitly because types nulls and undefined are all over the place.